engine/src/cmd/collision.cpp

```diff
@@ -156,96 +141,104 @@ void Collision::shouldApplyForceAndDealDamage(Unit* other_unit)
     }
 }
 
-// This stops the unit from going through the other unit
-/*void Collision::applyForce(double elasticity, float& m2, Vector& v2)
-{
-    //for torque... location -- approximation hack of MR^2 for rotational inertia (moment of inertia currently just M)
-    Vector torque = force/(unit->radial_size*unit->radial_size);
-    Vector force  = this->force-torque; // Note the variable shadowing
-
-    float  max_force  = max_force_multiplier*(unit->limits.forward + unit->limits.retro + unit->limits.lateral + unit->limits.vertical);
-    float  max_torque = max_torque_multiplier*(unit->limits.yaw + unit->limits.pitch+unit->limits.roll);
-
-    //Convert from frames to seconds, so that the specified value is meaningful
-    max_force  = max_force / (unit->sim_atom_multiplier * simulation_atom_var);
-    max_torque = max_torque / (unit->sim_atom_multiplier * simulation_atom_var);
-    float torque_magnitude = torque.Magnitude();
-    float force_magnitude = force.Magnitude();
-    if (torque_magnitude > max_torque) {
-        torque *= (max_torque/torque_magnitude);
-    }
-    if (force_magnitude > max_force) {
-        force *= (max_force/force_magnitude);
-    }
-    unit->ApplyTorque( torque, location );
-    unit->ApplyForce( force-torque );
-}*/
-
-
-
-
-void Collision::dealDamage(Collision other_collision, float factor)
+// Apply damage, in VS Damage units (a unit of energy), based on change in energy from collision outcome
+void Collision::dealDamage(Collision other_collision, double deltaKE_linear, double deltaKE_angular)
 {
     if(!deal_damage)
     {
         return;
     }
 
-    // Shorten variables for better visibility
-    float& m1 = mass;
-    float& m2 = other_collision.mass;
-    Vector& v1 = velocity;
-    Vector& v2 = other_collision.velocity;
-
-    float relative_velocity = (v1-v2).Magnitude();
-    float damage = relative_velocity * relative_velocity *
-            (m2)/(m1+m2) * factor;
-
+    float damage = 0.5 * ( deltaKE_linear + deltaKE_angular) /* deltaKE is in KiloJoules, due to mass being in units of 1000 Kg, not Kg - so convert accordingly */ / configuration.physics.kilojoules_per_damage; // assign half the change in energy to this unit, convert from KJ to VSD
```

Comment:
The `0.5` should be a constant declared somewhere; magic values shouldn't appear as anything else.

Reply:
Agreed - should factor this one out into a something representing the fraction of deltaKE assigned to this unit vs. the other unit.